catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/CatalogServiceImpl.java

```diff
@@ -415,6 +417,31 @@ CompletionStage<MultiTableUpdate> commit(ParsedReference reference, CatalogCommi
                 throw new RuntimeException(e);
               }
             })
+        // Add a failure handler, that cover the commitWithResponse() above but also all write
+        // failure that can happen in the stages added by
+        // applyIcebergTable/ViewCommitOperation().
+        .exceptionally(
+            e -> {
+              if (e instanceof CompletionException) {
+                e = e.getCause();
+              }
+              if (e.getClass() == RuntimeException.class
+                  && e.getMessage().equals(e.getCause().toString())) {
+                e = e.getCause();
+              }
```

Comment:
Why this? We don't do anything with the unwrapped exception. Also `exceptionally` is not designed to throw exceptions: I think `whenComplete` would be more appropriate here since you don't need to unwrap the exception.

Reply:
The code is just more boilerplate w/ `whenComplete` (null-check), but the exceptions are exactly the same - same unwrapping etc.
The handling for exceptions in `java.util.concurrent.CompletableFuture#uniExceptionally` and `java.util.concurrent.CompletableFuture#uniWhenComplete` is also the same.

Reply:
More boilerplate? I think it's less:

```java
        .whenComplete(
            (result, error) -> {
              if (error != null) {
                try {
                  objectIO.deleteObjects(
                      multiTableUpdate.storedLocations().stream()
                          .map(StorageUri::of)
                          .collect(toList()));
                } catch (Exception e) {
                  error.addSuppressed(e);
                }
              }
            })
```

Reply:
You still get a CompletionException there. The code is the same - but with the added `if (e != null)`.

Reply:
Try it - run TestCatalogServiceImpl w/ a breakpoint there - there's a CE

Reply:
And what is the problem with having a CE here? BTW the tests pass with `whenComplete`.

Reply:
The other way around: what's the problem with `exceptionally`, since the code's _less_ boilerplate because of the `if (e != null)`?

Reply:
Your proposal would wrap a `CompletionException` in another `CompletionException`, right?

Reply:
Oh - I see now. You're relyong on `addSuppressed` here